### PR TITLE
Upgrade to spring boot 2.2 and add prometheus endpoint

### PR DESCRIPTION
**Pull request by ties**
I'll run this for a while in my dev environment to make sure it works and resource usage is similar to last friday's branch.

Confirmed that test suite passes locally.

### CODE REVIEW

File: rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/TrustAnchorMetricsService.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    private ConcurrentHashMap<String, CertificateTreeValidationMetrics> cetrificateTreeValidationMetrics = new ConcurrentHashMap<>();
+
+    public void update(TrustAnchor ta, CertificateTreeValidationRun vr, long durationMs) {
+        final String uri = ta.getLocations().size() > 0 ? ta.getLocations().get(0) : null;
+        if (uri == null) {
+            log.error("Trust anchor {} does not have a location.", ta.getSubjectPublicKeyInfo());
+            return;
+        }
+
+        if (cetrificateTreeValidationMetrics.containsKey(uri)) {
```
Comment (lolepezy) [inline rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/TrustAnchorMetricsService.java]: Strictly speaking, ConcurrentHashMap doesn't protect from race conditions between calls of containsKey and get in this case. 

This should work atomically 
cetrificateTreeValidationMetrics
   .putIfAbsent(uri, new CertificateTreeValidationMetrics(ta, vr, durationMs))
   .update(vr, durationMs);

Comment (lolepezy) [reply rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/TrustAnchorMetricsService.java]: Or usual map and synchronised block.

Comment (ties) [reply rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/TrustAnchorMetricsService.java]: I was doing the initial update from the constructor. I made this slightly cleaner.

Comment (ties) [reply rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/TrustAnchorMetricsService.java]: It did not matter if the object was constructed twice. I cleaned this by moving the initial update from the constructor and doing that explicitly.

**Pull request merged**